### PR TITLE
Add GeocachingGroupingUICache project

### PR DESCRIPTION
**Pull request by Yodafon**

### CODE REVIEW

File: GeocachingGroupingUICache/build.gradle [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        }
+    }
+}
+
+dependencies {
+    implementation(project(':Shared'))
+    implementation('com.hazelcast:hazelcast:+')
+    implementation('org.springframework.boot:spring-boot-starter-web:2.7.0')
+    implementation('com.fasterxml.jackson.module:jackson-module-jaxb-annotations:+')
+
```
Comment (vilinet) [inline GeocachingGroupingUICache/build.gradle]: Empty lines!

File: GeocachingGroupingUICache/src/main/com/gg/uicache/GeocachingGroupingUICache.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+@SpringBootApplication
+@EnableJms
+@EnableAutoConfiguration
+@ComponentScan("com.gg.uicache")
+public class GeocachingGroupingUICache {
+
+    public static void main(String[] args) {
+        SpringApplication.run(GeocachingGroupingUICache.class);
+    }
+
```
Comment (vilinet) [inline GeocachingGroupingUICache/src/main/com/gg/uicache/GeocachingGroupingUICache.java:22]: Empty line!

File: GeocachingGroupingUICache/src/main/com/gg/uicache/GeocachingGroupingUICache.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        PropertySourcesPlaceholderConfigurer pspc
+                = new PropertySourcesPlaceholderConfigurer();
+        Resource[] resources = new ClassPathResource[]
+                {new ClassPathResource("/config/uicache.properties")};
+        pspc.setLocations(resources);
+        pspc.setIgnoreUnresolvablePlaceholders(true);
+        return pspc;
+    }
+
+
```
Comment (vilinet) [inline GeocachingGroupingUICache/src/main/com/gg/uicache/GeocachingGroupingUICache.java]: Empty line :P

File: GeocachingGroupingUICache/src/main/com/gg/uicache/config/JMSConfig.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.jms.support.converter.MessageType;
+
+@Configuration
+public class JMSConfig {
+
+    @Bean // Serialize message content to json using TextMessage
+    public MessageConverter jacksonJmsMessageConverter() {
+        MappingJackson2MessageConverter converter = new MappingJackson2MessageConverter();
+        converter.setTargetType(MessageType.TEXT);
+        converter.setTypeIdPropertyName("_type");
```
Comment (vilinet) [inline GeocachingGroupingUICache/src/main/com/gg/uicache/config/JMSConfig.java]: What is "_type"? Could be a constant with some reasoning why to be used.

File: GeocachingGroupingUICache/src/main/com/gg/uicache/incoming/JMSMessageListener.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Resource
+    private HazelcastInstance hazelcastInstance;
+
+    @JmsListener(destination = "${jms.county.input.queue}")
+    public void countyListener(GeocacheByCounty message) {
+        if (message.getRegion() != null) {
+            IMap<String, GeocacheByCounty> map = hazelcastInstance.getMap(HazelcastConfig.COUNTY_CACHE);
+            map.put(message.getCounty(), message);
+        }
```
Comment (vilinet) [inline GeocachingGroupingUICache/src/main/com/gg/uicache/incoming/JMSMessageListener.java]: Can't you have some common generic<T> function that does it for you and you can just call it from these Listener methods?

**Review by vilinet — CHANGES_REQUESTED** [review]

File: GeocachingGroupingUICache/src/main/com/gg/uicache/incoming/JMSMessageListener.java (hunk 5 above)
Comment (Yodafon) [reply GeocachingGroupingUICache/src/main/com/gg/uicache/incoming/JMSMessageListener.java]: fixed

File: GeocachingGroupingUICache/src/main/com/gg/uicache/config/JMSConfig.java (hunk 4 above)
Comment (Yodafon) [reply GeocachingGroupingUICache/src/main/com/gg/uicache/config/JMSConfig.java:16]: fixed

File: GeocachingGroupingUICache/src/main/com/gg/uicache/GeocachingGroupingUICache.java (hunk 3 above)
Comment (Yodafon) [reply GeocachingGroupingUICache/src/main/com/gg/uicache/GeocachingGroupingUICache.java]: fixed

File: GeocachingGroupingUICache/src/main/com/gg/uicache/GeocachingGroupingUICache.java (hunk 2 above)
Comment (Yodafon) [reply GeocachingGroupingUICache/src/main/com/gg/uicache/GeocachingGroupingUICache.java:22]: fixed

File: GeocachingGroupingUICache/build.gradle (hunk 1 above)
Comment (Yodafon) [reply GeocachingGroupingUICache/build.gradle]: fixed

**Review by vilinet — APPROVED** [review]

File: GeocachingGroupingUICache/src/main/com/gg/uicache/config/JMSConfig.java (hunk 4 above)
Comment (Yodafon) [reply GeocachingGroupingUICache/src/main/com/gg/uicache/config/JMSConfig.java]: fixed

**Pull request merged**